### DISCUSSION TITLE
Dynamically add or remove Tab component，may cause compoment's selected status abnormally

### DISCUSSION THREAD

**Discussion by mahiru-forever** [carbon-design-system/carbon-components-svelte#1494] [General]
Whatever i use the selected(Tabs component's props) or not, it always operation abnormally.I saw the code, Tabs Component's tabs variable seem only perceive the newly added Tab component, but when delete, tabs is no change. I'm not sure it's my code wrong or not used correctly.
Look forward to your answer.

```js
// tabs component template
<Tabs>
  {#each tabs as { label, key }}
  <Tab>
     {label}
       <span class="del-tab" on:click|preventDefault|stopPropagation={() => delMehod(key)}>x</span>
  </Tab>
  {/each}
  <div class="tab-add" on:click={addMethod}>+</div>
</Tabs>

// dynamically remove
function delMehod(key) {
  tabs = tabs.filter(v => {
    return v.key !== el.detail
  })
}

// dynamically add
function addMethod() {
  tabs = [...tabs, { label:'xxx', key: 'xxx' }]
}
```

**Comment by metonym** [comment]
Duplicate of #368

`Tabs` uses component composition and relies on the index for ordering. If removed or re-ordered, the index is lost.

The workaround is to use the [`#key` block](https://svelte.dev/docs#template-syntax-key) to re-create the component any time `tabs` changes:

```html
{#key tabs}
  <Tabs>
     {#each tabs as tab}
       ...
     {/each}
  </Tabs>
{/key
```

**Reply by thangvd2312** [reply]
when i use {#key} it will re-render the components in the content of tabs, i don't want that to happen, is there any way to fix that, thanks

**Comment by metonym** [comment]
Fixed in [v0.93.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.93.0)

Example: https://svelte.carbondesignsystem.com/components/Tabs#conditional-tabs
